main.py

```diff
@@ -60,3 +60,52 @@ def infinite_loop():
         if len(w) > 4:
             words.insert(0, w)
     return words
+
+
+# Prefer DefaultDict over setDefult
+def setdefault_example():
+    std_dict = dict()
+    for k, v  in enumerate(range(5)):
+        std_dict.setdefault(k, []).append(v)
+    return std_dict
+
+# This method reads multiple file paths, open each file to load data, but forgets to close it, leading to resource leaks
+# Further, it selectively processes content based on string find condition. The find() operation can be simply replaced
+# with a membership tests because one does not need to know the position at which the search keyword appears.
+def process_kinesis(kinesis, file_list_path):
+    flp = open(file_list_path, 'r')
+    for line in flp:
+        file_path = line.strip('\n').strip('\r\n')
+        fp = open(file_path, 'r')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fp*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *next()* throws an exception. To prevent this resource leak, close *fp* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).